src/Enum.php

```diff
@@ -37,6 +37,14 @@ abstract class Enum implements \JsonSerializable
      */
     protected static $cache = [];
```

Comment:
What about this cache?

Reply:
This is a cache of the allowed string values of an Enum (from reflection), this cache is still needed and used. The cache in this pull is a map of constant names ->`Enum` instances so we don't have to construct a new instance each time.